File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/domain/Orders/Models/Registrant.cs
using System;
using ECommon.Utilities;

namespace Registration.Orders
{
    public class Registrant
    {
        public string FirstName { get; private set; }
        public string LastName { get; private set; }
        public string Email { get; private set; }

        public Registrant() { }
        public Registrant(string firstName, string lastName, string email)
        {
            Ensure.NotNull(firstName, "firstName");
            Ensure.NotNull(lastName, "lastName");
            Ensure.NotNull(email, "email");
            FirstName = firstName;
            LastName = lastName;
            Email = email;
        }
    }
}

File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/domain/orders/Models/Registrant.java
package com.microsoft.conference.registration.domain.orders.Models;

import org.enodeframework.common.utilities.Ensure;

public class Registrant {
    public String firstName;
    public String lastName;
    public String email;

    public Registrant() {
    }

    public Registrant(String firstName, String lastName, String email) {
        Ensure.notNull(firstName, "firstName");
        Ensure.notNull(lastName, "lastName");
        Ensure.notNull(email, "email");
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
    }
}
